client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/DeleteWatchRequest.java

```diff
@@ -58,28 +55,14 @@ public void setId(String id) {
     }
 
     @Override
-    public ActionRequestValidationException validate() {
-        ActionRequestValidationException validationException = null;
-        if (id == null){
-            validationException = ValidateActions.addValidationError("watch id is missing", validationException);
+    public Optional<ValidationException> validate() {
```

Comment:
lets put this in the constructor instead if possible.

https://github.com/elastic/elasticsearch/blob/master/client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/DeactivateWatchRequest.java#L29-L37